File: Day245/sticklerThief.go
package main

import (
	"fmt"
	"math"
)

func findMaxSum(arr []int, n int) int {
	curr , prev := 0, 0
	for _,v := range arr {
		temp := int(math.Max(float64(prev + v), float64(curr)))
		prev = curr
		curr = temp
	}
	return curr
}

func main() {
	arr := []int{6,5,5,7,4}

	fmt.Print(findMaxSum(arr, 5))
}

File: Day245/sticklerThief.java
package Day245;

public class sticklerThief {

    public static int findMaxSum(int arr[], int n) {
        int curr = 0, prev = 0;
        for (int i=0; i<n; i++) {
            int temp = Math.max(prev + arr[i], curr);
            prev = curr;
            curr = temp;
        }
        return curr;
    }

    public static void main(String[] args) {
        int arr[] = {6,5,5,7,4};
        System.out.println(findMaxSum(arr, 5));

    }
}